semantic-ast/docs/json-ASTs.md

```diff
@@ -0,0 +1,42 @@
+**⚠️WIP:⚠️** _I'm working through the open questions listed below and will move them up into the Design section as I answer them._
+
+# JSON output for Python
+
+`semantic-ast` is dedicated to the production of ASTs. The package provides a CLI for auto-generating ASTs from source code. This is the output of the `parseByteString` utility in `haskell-tree-sitter`. Currently, we only parse Python source, but this will soon change as we add languages. In addition to supporting more languages, consumers of this library should also be able to specify the output format in which they'd like to display generated ASTs. JSON is one of such formats, and one we can prioritize given its standard usage across a variety of applications and use-cases.
+
+This document explores how to produce JSON with the hope that it will a) clarify format-specific requirements and b) help us explore the broader design space involving AST outputs across multiple languages and multiple formats. This exercise will likely inform our opinions for the best way to generalize the approach and proceed for other formats (ex., `s-expression` etc).
+
+### Table of Contents
+
+- [Use-cases](#use-cases)
+- [Possible user-facing features/endpoints](#possible-user-facing-features-endpoints)
+- [Design]
+- [Open questions](open-questions)
+
+### Use-cases
+
+Exploring how this JSON would be consumed informs the shape of the output we produce (ex., most likely one JSON object per AST node). Here are a few use-cases that come to mind:
+- External user wants to verify something about an AST, `semantic-ast` gives user an AST that can be piped into `jq` and be used to slice data in various ways
+- External user wants to process the code through _some_ research pipeline
+- GitHub action that parses your code and verifies some property (like a linter)
+- ML team: adjacency list not tree
+- Mobile team: mobile Code Nav features (though this is something to explore using the current pipeline in place, _not_ just precise ASTs)
```

Comment:
As the parenthetical hints at, I think that mobile code nav should be out of scope for this document. Properly producing code nav on mobile will require implementing endpoints in the main github application so that we can properly leverage authN/Z.

Reply:
Yes! I glommed it all in here but they are two very separate streams. This involves precise ASTs, and the mobile graphQL endpoint is more about figuring out how to structure existing (imprecise) ASTs.